docs/src/api-testing-python.md

```diff
@@ -0,0 +1,289 @@
+title: "API testing"
+---
+
+Playwright can be used to get access to the [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) API of
+your application.
+
+Sometimes you may want to send requests to the server directly from Node.js without loading a page and running js code in it.
+A few examples where it may come in handy:
+- Test your server API.
+- Prepare server side state before visiting the web application in a test.
+- Validate server side post-conditions after running some actions in the browser.
+
+All of that could be achieved via [APIRequestContext] methods.
+
+The following examples rely on the [`pytest-playwright`](./test-runners.md) package which add Playwright fixtures to the Pytest test-runner.
+
+<!-- TOC -->
+
+## Writing API Test
+
+[APIRequestContext] can send all kinds of HTTP(S) requests over network.
+
+The following example demonstrates how to use Playwright to test issues creation via [GitHub API](https://docs.github.com/en/rest). The test suite will do the following:
+- Create a new repository before running tests.
+- Create a few issues and validate server state.
+- Delete the repository after running tests.
+
+### Configure
+
+GitHub API requires authorization, so we'll configure the token once for all tests. While at it, we'll also set the `baseURL` to simplify the tests.
+
+```python
+import os
+from typing import Generator
+
+import pytest
+from playwright.sync_api import Playwright, APIRequestContext
+
+GITHUB_API_TOKEN = os.getenv("GITHUB_API_TOKEN")
+assert GITHUB_API_TOKEN, "GITHUB_API_TOKEN is not set"
+
+
+@pytest.fixture(scope="session")
+def api_request_context(
+    playwright: Playwright,
+) -> Generator[APIRequestContext, None, None]:
+    headers = {
+        # We set this header per GitHub guidelines.
+        "Accept": "application/vnd.github.v3+json",
+        # Add authorization token to all requests.
+        # Assuming personal access token available in the environment.
+        "Authorization": f"token {GITHUB_API_TOKEN}",
+    }
+    request_context = playwright.request.new_context(
+        base_url="https://api.github.com", extra_http_headers=headers
+    )
+    yield request_context
+    request_context.dispose()
+
+```
+
+### Write tests
+
+Now that we initialized request object we can add a few tests that will create new issues in the repository.
+```python
+import os
+from typing import Generator
+
+import pytest
+from playwright.sync_api import Playwright, APIRequestContext
+
+GITHUB_API_TOKEN = os.getenv("GITHUB_API_TOKEN")
+assert GITHUB_API_TOKEN, "GITHUB_API_TOKEN is not set"
+
+GITHUB_USER = os.getenv("GITHUB_USER")
+assert GITHUB_USER, "GITHUB_USER is not set"
+
+GITHUB_REPO = "test"
+
+# ...
+
+def test_should_create_bug_report(api_request_context: APIRequestContext) -> None:
+    data = {
+        "title": "[Bug] report 1",
+        "body": "Bug description",
+    }
+    new_issue = api_request_context.post(f"/repos/{GITHUB_USER}/{GITHUB_REPO}/issues", data=data)
```

Comment:
Are you planning to add async versions too?

Reply:
I'd only do sync for now.